### ISSUE TITLE
Basic Details update failed problem

### ISSUE THREAD

**Issue by YASSERKOT** [sharetribe/sharetribe#3012]
**##### Sharetribe Version:** 
          6.4.0
**##### Environment:**
          Ruby 2.3  || Ruby on Rails 5.1.1
**##### Linux distribution :** 
          Ubuntu 16.04.
**##### Node version:**
> node -v 
v 8.1.3
> npm -v  
5.0.3

**##### Server mode:**
I am running under development mode.
**##### Description:**
I have run the sharetribe on my localhost and I have faced a problem with updating the Basic details at my admin panel .
I think the problem is happening because the "Enabled languages" bar dosen't show up but I didn't know how to fix it !!
When I click on "save settings" it show me an error message above saying : "Detail update failed."

**Comment by bladealslayer** [comment]
@YASSERKOT what does the application log contain?

**Comment by YASSERKOT** [comment]
For my console log, it shows me as follow : 

` **14:18:49 rails.1**               | INFO [react_on_rails] RENDERED TopbarApp to dom node with id: topbar-container with railsContext:","{\"inMailer\":false,\"i18nLocale\":\"en\",\"i18nDefaultLocale\":\"en\",\"href\":\"http://localhost:3000/en/admin/details/edit\",\"location\":\"/en/admin/details/edit\",\"scheme\":\"http\",\"host\":\"localhost\",\"port\":3000,\"pathname\":\"/en/admin/details/edit\",\"search\":null,\"httpAcceptLanguage\":\"en-US,en;q=0.5\",\"marketplaceId\":1,\"loggedInUsername\":\"yasserk\",\"marketplace_color1\":\"#4a90e2\",\"marketplace_color2\":\"#2ab865\",\"marketplace_slogan_color\":\"#ffffff\",\"marketplace_description_color\":\"#ffffff\",\"serverSide\":true}

**14:18:49 rails.1**               | INFO {"method":"GET","path":"/en/admin/details/edit","format":"html","controller":"admin/community_customizations","action":"edit_details","status":200,"duration":342.19,"view":187.42,"db":97.28,"params":{"locale":"en"},"host":"localhost","community_id":1,"current_user_id":"F6dMf-xtub9vMKT_3gCy8g","user_agent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:54.0) Gecko/20100101 Firefox/54.0","referer":"http://localhost:3000/en/admin/getting_started_guide/slogan_and_description","forwarded_for":"","request_uuid":"8fcfc60a-eb8b-4375-96c8-118e544b0e9b"}

**14:18:49 rails.1**               | INFO {"method":"GET","path":"/en/admin/details/edit","format":"html","controller":"admin/community_customizations","action":"edit_details","status":200,"duration":342.19,"view":187.42,"db":97.28,"params":{"locale":"en"},"host":"localhost","community_id":1,"current_user_id":"F6dMf-xtub9vMKT_3gCy8g","user_agent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:54.0) Gecko/20100101 Firefox/54.0","referer":"http://localhost:3000/en/admin/getting_started_guide/slogan_and_description","forwarded_for":"","request_uuid":"8fcfc60a-eb8b-4375-96c8-118e544b0e9b"}

**14:18:49 rails.1**               | DEBUG    (2.6ms)  SELECT COUNT(*) FROM `communities`

**14:18:49 rails.1**               | DEBUG   Community Load (0.4ms)  SELECT  `communities`.* FROM `communities`  ORDER BY `communities`.`id` ASC LIMIT 1

**14:18:49 rails.1**               | DEBUG   Person Load (0.5ms)  SELECT  `people`.* FROM `people` WHERE `people`.`id` = 'F6dMf-xtub9vMKT_3gCy8g'  ORDER BY `people`.`id` ASC LIMIT 1

**14:18:49 rails.1**               | DEBUG   CommunityMembership Load (0.3ms)  SELECT  `community_memberships`.* FROM `community_memberships` WHERE `community_memberships`.`person_id` = 'F6dMf-xtub9vMKT_3gCy8g' LIMIT 1

**14:18:49 rails.1**               | DEBUG   FeatureFlag Load (0.5ms)  SELECT `feature_flags`.* FROM `feature_flags` WHERE (community_id = 1 AND (person_id IS NULL OR person_id = 'F6dMf-xtub9vMKT_3gCy8g'))

**14:18:49 rails.1**               | DEBUG   CommunityMembership Load (0.3ms)  SELECT  `community_memberships`.* FROM `community_memberships` WHERE `community_memberships`.`person_id` = 'F6dMf-xtub9vMKT_3gCy8g' AND `community_memberships`.`community_id` = 1 AND `community_memberships`.`status` = 'accepted'  ORDER BY `community_memberships`.`id` ASC LIMIT 1

**14:18:49 rails.1**               | DEBUG    (0.2ms)  BEGIN

**14:18:49 rails.1**               | DEBUG   SQL (0.3ms)  INSERT INTO `delayed_jobs` (`handler`, `queue`, `priority`, `run_at`, `created_at`, `updated_at`) VALUES ('--- !ruby/struct:PageLoadedJob\ncommunity_membership_id: 1\nhost: localhost\n', 'default', 5, '2017-07-12 13:18:49', '2017-07-12 13:18:49', '2017-07-12 13:18:49')

**14:18:49 rails.1**               | DEBUG    (78.3ms)  COMMIT

**14:18:49 rails.1**               | DEBUG   CommunityCustomization Load (0.4ms)  SELECT  `community_customizations`.* FROM `community_customizations` WHERE `community_customizations`.`community_id` = 1 AND `community_customizations`.`locale` = 'en'  ORDER BY `community_customizations`.`id` ASC LIMIT 1

**14:18:49 rails.1**               | DEBUG   CommunityCustomization Load (0.3ms)  SELECT  `community_customizations`.* FROM `community_customizations` WHERE `community_customizations`.`community_id` = 1 AND `community_customizations`.`locale` = 'en'  ORDER BY `community_customizations`.`id` ASC LIMIT 1

**14:18:49 rails.1**               | DEBUG   PaymentSettings Load (0.3ms)  SELECT  `payment_settings`.* FROM `payment_settings` WHERE `payment_settings`.`community_id` = 1 AND `payment_settings`.`active` = 1  ORDER BY `payment_settings`.`id` ASC LIMIT 1

**14:18:49 rails.1**               | DEBUG   CommunityCustomization Load (0.5ms)  SELECT  `community_customizations`.* FROM `community_customizations` WHERE `community_customizations`.`community_id` = 1 AND `community_customizations`.`locale` = 'en' LIMIT 1

**14:18:49 rails.1**               | DEBUG   TransactionProcess Load (0.3ms)  SELECT `transaction_processes`.* FROM `transaction_processes` WHERE `transaction_processes`.`community_id` = 1

**14:18:49 rails.1**               | DEBUG   MarketplaceSetupSteps Load (0.3ms)  SELECT  `marketplace_setup_steps`.* FROM `marketplace_setup_steps` WHERE `marketplace_setup_steps`.`community_id` = 1 LIMIT 1

**14:18:49 rails.1**               | DEBUG   CACHE (0.0ms)  SELECT  `payment_settings`.* FROM `payment_settings` WHERE `payment_settings`.`community_id` = 1 AND `payment_settings`.`active` = 1  ORDER BY `payment_settings`.`id` ASC LIMIT 1  [["community_id", 1], ["active", true]]

**14:18:49 rails.1**               | DEBUG   CACHE (0.0ms)  SELECT  `community_customizations`.* FROM `community_customizations` WHERE `community_customizations`.`community_id` = 1 AND `community_customizations`.`locale` = 'en'  ORDER BY `community_customizations`.`id` ASC LIMIT 1  [["community_id", 1], ["locale", :en]]

**14:18:49 rails.1**               | DEBUG   CACHE (0.0ms)  SELECT  `community_customizations`.* FROM `community_customizations` WHERE `community_customizations`.`community_id` = 1 AND `community_customizations`.`locale` = 'en'  ORDER BY `community_customizations`.`id` ASC LIMIT 1  [["community_id", 1], ["locale", :en]]

**14:18:49 rails.1**               | DEBUG   CACHE (0.0ms)  SELECT  `community_customizations`.* FROM `community_customizations` WHERE `community_customizations`.`community_id` = 1 AND `community_customizations`.`locale` = 'en'  ORDER BY `community_customizations`.`id` ASC LIMIT 1  [["community_id", 1], ["locale", :en]]

**14:18:49 rails.1**               | DEBUG   CACHE (0.0ms)  SELECT  `community_customizations`.* FROM `community_customizations` WHERE `community_customizations`.`community_id` = 1 AND `community_customizations`.`locale` = 'en'  ORDER BY `community_customizations`.`id` ASC LIMIT 1  [["community_id", 1], ["locale", :en]]

**14:18:49 rails.1**               | DEBUG   CACHE (0.0ms)  SELECT  `community_customizations`.* FROM `community_customizations` WHERE `community_customizations`.`community_id` = 1 AND `community_customizations`.`locale` = 'en'  ORDER BY `community_customizations`.`id` ASC LIMIT 1  [["community_id", 1], ["locale", :en]]

**14:18:49 rails.1**               | DEBUG   CACHE (0.0ms)  SELECT  `marketplace_setup_steps`.* FROM `marketplace_setup_steps` WHERE `marketplace_setup_steps`.`community_id` = 1 LIMIT 1  [["community_id", 1]]

**14:18:49 rails.1**               | INFO [react_on_rails] RENDERED OnboardingTopBar to dom node with id: OnboardingTopBar-react-component-34432337-ea7a-4b97-bdbc-d7a2817fb7c7 with railsContext:","{\"inMailer\":false,\"i18nLocale\":\"en\",\"i18nDefaultLocale\":\"en\",\"href\":\"http://localhost:3000/en/admin/details/edit\",\"location\":\"/en/admin/details/edit\",\"scheme\":\"http\",\"host\":\"localhost\",\"port\":3000,\"pathname\":\"/en/admin/details/edit\",\"search\":null,\"httpAcceptLanguage\":\"en-US,en;q=0.5\",\"marketplaceId\":1,\"loggedInUsername\":\"yasserk\",\"marketplace_color1\":\"#4a90e2\",\"marketplace_color2\":\"#2ab865\",\"marketplace_slogan_color\":\"#ffffff\",\"marketplace_description_color\":\"#ffffff\",\"serverSide\":true}

**14:18:49 rails.1**               | INFO [react_on_rails] RENDERED OnboardingTopBar to dom node with id: OnboardingTopBar-react-component-34432337-ea7a-4b97-bdbc-d7a2817fb7c7 with railsContext:","{\"inMailer\":false,\"i18nLocale\":\"en\",\"i18nDefaultLocale\":\"en\",\"href\":\"http://localhost:3000/en/admin/details/edit\",\"location\":\"/en/admin/details/edit\",\"scheme\":\"http\",\"host\":\"localhost\",\"port\":3000,\"pathname\":\"/en/admin/details/edit\",\"search\":null,\"httpAcceptLanguage\":\"en-US,en;q=0.5\",\"marketplaceId\":1,\"loggedInUsername\":\"yasserk\",\"marketplace_color1\":\"#4a90e2\",\"marketplace_color2\":\"#2ab865\",\"marketplace_slogan_color\":\"#ffffff\",\"marketplace_description_color\":\"#ffffff\",\"serverSide\":true}

**14:18:49 rails.1**               | DEBUG   MenuLink Load (0.3ms)  SELECT `menu_links`.* FROM `menu_links` WHERE `menu_links`.`community_id` = 1  ORDER BY sort_priority

**14:18:49 rails.1**               | DEBUG    (0.2ms)  SELECT `landing_pages`.`enabled`, `landing_pages`.`released_version` FROM `landing_pages` WHERE `landing_pages`.`community_id` = 1
`
Also the "sharetribe/development.searchd.log" file contains : 
`[Wed Jul  5 20:00:18.024 2017] [27117] watchdog: main process 27118 forked ok
[Wed Jul  5 20:00:18.025 2017] [27118] listening on all interfaces, port=3563
[Wed Jul  5 20:00:18.030 2017] [27118] accepting connections
[Wed Jul  5 21:54:20.622 2017] [27118] caught SIGHUP (seamless=1, in queue=1)
[Wed Jul  5 21:54:20.742 2017] [27119] rotating index 'listing_delta': started
[Wed Jul  5 21:54:20.807 2017] [27119] rotating index 'listing_delta': success
[Wed Jul  5 21:54:20.817 2017] [27119] rotating index 'listing_core': started
[Wed Jul  5 21:54:20.819 2017] [27119] rotating index 'listing_core': success
[Wed Jul  5 21:54:20.821 2017] [27119] rotating index 'custom_field_value_delta': started
[Wed Jul  5 21:54:20.823 2017] [27119] rotating index 'custom_field_value_delta': success
[Wed Jul  5 21:54:20.823 2017] [27119] rotating index 'custom_field_value_core': started
[Wed Jul  5 21:54:20.825 2017] [27119] rotating index 'custom_field_value_core': success
[Wed Jul  5 21:54:20.826 2017] [27119] rotating index: all indexes done
[Thu Jul  6 19:07:39.810 2017] [27118] caught SIGHUP (seamless=1, in queue=1)
[Thu Jul  6 19:07:41.582 2017] [27118] caught SIGTERM, shutting down
[Thu Jul  6 19:07:53.192 2017] [27118] shutdown complete`

**Comment by bladealslayer** [comment]
I'm guessing that something went wrong when the marketplace was being first initialised when you filled the marketplace form after first starting the development environment. Could you wipe the database and start over and see if you get the same issue? If you do, can you post screenshot of how the basic details page looks like?

**Comment by YASSERKOT** [comment]
I recreated the database and reinitialized its content and nothing was been settled and the issue persists.
![git1](https://user-images.githubusercontent.com/19874315/28314304-433617f6-6bba-11e7-84cc-b3ec78bf8198.png)
As you see the language box selector isn't shown up on the page and ofc when i submit the form without being able to pick a value, the marketplace returns error message.
Also I have noticed that the post listing page isn't working and this what it shows : 
![screenshot from 2017-07-18 12-07-53](https://user-images.githubusercontent.com/19874315/28314494-0f07819e-6bbb-11e7-94af-51244495875b.png)

I tried reinstalling node.js and react modules another time and all the package are being installed but the issue wasn't resolved.

**Comment by mickeyren** [comment]
same here - i'm suffering with this kind of error on `63cf226`. the data persists, i was able to fill all the fields, but the marketplace percentage completion is stuck at 28%. it's not moving.

**Comment by jkadry** [comment]
I'm also having this issue on development environment. Tried reinstalling nodes and wiping db, but issue still persists. Anyone have any other idea what might be causing this?

**Comment by mickeyren** [comment]
i was able to successfully start the app with node 6.9.5. Try this node version?

**Comment by bladealslayer** [comment]
Indeed, I didn't notice the incorrect node version. Sharetribe 6.4.0 requires node 6.9.x. I'm going to close this.

**Comment by kriskoli** [comment]
I have the same issue on local developwith mode with sharetribe 7.0. It needs node version 7.8 and npm version 4.2.

**Comment by tustis42** [comment]
Same issue for me:
javi@:~/sharetribe$ nvm ls
->       v7.8.0
         system
default -> 7.8.0 (-> v7.8.0)
node -> 7.8.0 (-> v7.8.0)
stable -> 7.8.0 (-> v7.8.0)
iojs -> N/A (default)
lts/* -> lts/boron (-> N/A)
lts/argon -> v4.8.4 (-> N/A)
lts/boron -> v6.11.2 (-> N/A)
javi@:~/sharetribe$ npm -v
4.2.0
javi@:~/sharetribe$ cat VERSION
7.0.0

Anyone to give a tip on what can be wrong?

**Comment by terrence0488** [comment]
Any updates??
Having the same problem here.
Using:
  Sharetribe version 7.1
  node 7.8.0
  npm 4.2.0

No error from server side log. After clicking the "Save Setting" button, server output showing a PATCH request with status 302. But the red bar "Detail update failed." just won't go away.

![screenshot](https://preview.ibb.co/eMSmB5/Screen_Shot_2017_09_25_at_2_16_45_AM.png "screenshot")

not sure if it's something related.

**Comment by sophy17** [comment]
Hello
I hiverne same issue
Any updates??
Having the same problem here.
Using on :
Ubuntu 16.04 x64
in Production mode
Sharetribe version 7.1
node 7.8.0
npm 4.2.0

I try to remove sharetribe,  delete the db and the problem is always the some 
Cab you help mz please 
Urgent

**Comment by hounsellal** [comment]
I had the same problem. The problem was that I went through the installation instructions using node v10 (because I didn't read all the steps or figured any version of node would work). This included running npm install while using v10. Then it didn't run so I used nvm to use v7.8.0 and it would run, but I had this same red error box with "Detail update failed" on the first on boarding page.

The solution was:

1. Delete the folder client/node_modules
2. Ensure I'm using 7.8.0 with node --version. If not, then: nvm use 7.8.0
3. Go back into client and run "npm i"

Everything works fine now.
